plugins/Live/API.php

```diff
@@ -189,6 +189,10 @@ public function getVisitorProfile($idSite, $visitorId = false, $segment = false,
     {
         Piwik::checkUserHasViewAccess($idSite);
 
+        if (!Live::isVisitorProfileEnabled($idSite)) {
+            throw new Exception('Visitor profile has been disabled in website settings');
+        }
```

Comment:
Should this be translated? Since the user could see it.

Reply:
He should actually only see this when calling the API directly. The UI shouldn't try to load the visitor profile at all when it's disabled

Reply:
I can't remember how, but I was able to see the error in the UI a couple times. Once in widgets, once in the visitor log page. This was while switching the setting on/off in one tab and testing in another, however, so it could have been an edge case users wouldn't notice.